connectors/kafka/src/main/java/forklift/connectors/KafkaConnector.java

```diff
@@ -0,0 +1,167 @@
+
+    public KafkaConnector(String kafkaHosts, String schemaRegistries, String groupId) {
+        this.kafkaHosts = kafkaHosts;
+        this.schemaRegistries = schemaRegistries;
+        this.groupId = groupId;
+    }
+
+    @Override
+    public void start() throws ConnectorException {
+
+        Properties producerProperties = new Properties();
+        producerProperties.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaHosts);
+        producerProperties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+                               io.confluent.kafka.serializers.KafkaAvroSerializer.class);
+        producerProperties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+                               io.confluent.kafka.serializers.KafkaAvroSerializer.class);
+        //schema.registry.url is a comma separated list of urls
+        producerProperties.put("schema.registry.url", schemaRegistries);
+        this.kafkaProducer = new KafkaProducer(producerProperties);
+        this.controller = createController();
+        this.controller.start();
+    }
+
+    private KafkaController createController() {
+        Properties props = new Properties();
+        props.put("bootstrap.servers", kafkaHosts);
+        props.put("group.id", groupId);
+        props.put("enable.auto.commit", false);
+        props.put("consumer.timeout.ms", "-1");
+        props.put("key.deserializer", io.confluent.kafka.serializers.KafkaAvroDeserializer.class);
+        props.put("value.deserializer", io.confluent.kafka.serializers.KafkaAvroDeserializer.class);
+        props.put("schema.registry.url", schemaRegistries);
+        props.put("specific.avro.reader", false);
+        //props.put("auto.offset.reset", "earliest");
+        this.kafkaConsumer = new KafkaConsumer(props);
+        this.controller = new KafkaController(kafkaConsumer, messageStream);
+        return controller;
+    }
+
+    @Override
+    public void stop() throws ConnectorException {
+        try {
+            this.controller.stop(2000, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+            log.error("KafkaConnector interrupted while stopping");
+        }
+        this.kafkaProducer.close();
+    }
+
+    @Override
+    public Connection getConnection() throws ConnectorException {
+        throw new UnsupportedOperationException("Kafka does not support Connections, please work with Queues and Topics");
+    }
+
+    @Override
+    public ForkliftConsumerI getQueue(String name) throws ConnectorException {
+        return this.getTopic(name);
+    }
+
+    @Override
+    public ForkliftConsumerI getTopic(String name) throws ConnectorException {
+        synchronized (this) {
+            if (this.controller == null || !this.controller.isRunning()) {
+                this.controller = createController();
+                this.controller.start();
+            }
+        }
+        return new KafkaTopicConsumer(name, controller, messageStream);
```

Comment:
Design nit: I think it would make more sense to have the controller manage the details of creating a consumer. That way you can put the logic for setting up a consumer to use a controller in the controller rather than having the consumer manage state to keep track of whether the controller is actually completely set up for it. This also has the minor benefit of avoiding exposing the `MessageStream` object to the connector if it doesn't need to be exposed.